File: challenge.ts
//import inputs.txt file to typescript
import * as fs from "fs";

//read the txt file
let fileContent = fs.readFileSync("inputs.txt", "utf8");

//split the content in inputs.txt by whitespace and parse the strings to intergers
const inputs = fileContent.split("\n").map((line) => parseInt(line, 10));

// //function to calculate the fuel
const calculation = (num: number) => {
  return Math.floor(num / 3) - 2;
};

//calculate by using the function above for each number
for (let i = 0; i < inputs.length; i++) {
  inputs[i] = calculation(inputs[i]);
}

//calculate the total number all the results
const sum = (nums: number[]): number => {
  let total = 0;
  for (let i = 0; i < nums.length; i++) {
    total += nums[i];
  }
  return total;
};

//show to result in the terminal
console.log(sum(inputs));

File: challenge.py
import math

# Open the 'inputs.txt' file for reading
InputFromFile = open("./inputs.txt", "r")

# Initialize an empty list to store the numbers
Inputs = []

# Initialize the variable to accumulate the total sum of results
Sum = 0

# Read each line from the file, strip whitespace, and convert to an integer
for line in InputFromFile:
    Inputs.append(int(line.strip()))

# Function to calculate the fuel requirement


def Equation(x):
    num = math.floor(x / 3) - 2
    return num


# Calculate the total of all results by applying the calculation function
for x in Inputs:
    Sum += Equation(x)

# Print the total sum to the terminal
print(Sum)
